scrapy/downloadermiddlewares/cookies.py

```diff
@@ -81,6 +82,14 @@ def _format_cookie(self, cookie):
         return cookie_str
 
     def _get_request_cookies(self, jar, request):
+        # from 'Cookie' request header
+        cookie_header = request.headers.get('Cookie', '')
+        cookie_list = re.split(';\s*', cookie_header.decode('latin1'))
```

Comment:
Are there stdlib functions to parse cookies?

Also, cookies should be UTF-8 in most cases (but other encodings are also possible, so the code must be robust); .decode('latin1') never fails, but I wonder if non-ascii UTF-8 cookies are preserved correctly now. Could you please add a test for it?